contracts/src/RealitioForeignArbitrationProxyWithAppeals.sol

```diff
@@ -0,0 +1,602 @@
+
+        uint256 arbitrationCost = arbitrator.arbitrationCost(arbitratorExtraData);
+        require(msg.value >= arbitrationCost, "Deposit value too low");
+
+        arbitration.status = Status.Requested;
+        arbitration.requester = msg.sender;
+        arbitration.deposit = msg.value;
+
+        bytes4 methodSelector = IHomeArbitrationProxy(0).receiveArbitrationRequest.selector;
+        bytes memory data =
+            abi.encodeWithSelector(methodSelector, bytes32(_arbitrationID), bytes32(_contestedAnswer), msg.sender);
+        amb.requireToPassMessage(homeProxy, data, amb.maxGasPerTx());
+
+        emit ArbitrationRequested(_arbitrationID, _contestedAnswer, msg.sender);
+    }
+
+    /**
+     * @notice Creates a dispute for a given question ID.
+     * @param _questionID The ID of the question.
+     */
+    function acknowledgeArbitration(bytes32 _questionID) external override onlyAmb onlyHomeChain onlyHomeProxy {
+        Arbitration storage arbitration = arbitrations[uint256(_questionID)];
+        require(arbitration.status == Status.Requested, "Invalid arbitration status");
+
+        uint256 arbitrationCost = arbitrator.arbitrationCost(arbitratorExtraData);
+
+        if (arbitration.deposit < arbitrationCost) {
+            arbitration.status = Status.Failed;
+
+            emit ArbitrationFailed(uint256(_questionID));
+        } else {
+            // At this point, arbitration.deposit is guaranteed to be greater than or equal to the arbitration cost.
+            uint256 remainder = arbitration.deposit - arbitrationCost;
+
+            uint256 disputeID =
+                arbitrator.createDispute{value: arbitrationCost}(NUMBER_OF_CHOICES_FOR_ARBITRATOR, arbitratorExtraData);
+            externalIDtoLocalID[disputeID] = uint256(_questionID);
+            arbitration.status = Status.Created;
+            arbitration.deposit = 0;
+            arbitration.disputeID = disputeID;
+            arbitration.rounds.push();
+
+            if (remainder > 0) {
+                arbitration.requester.send(remainder);
+            }
+
+            emit ArbitrationCreated(uint256(_questionID), disputeID);
+        }
+    }
+
+    /**
+     * @notice Cancels the arbitration.
+     * @param _questionID The ID of the question.
+     */
+    function cancelArbitration(bytes32 _questionID) external override onlyAmb onlyHomeChain onlyHomeProxy {
+        Arbitration storage arbitration = arbitrations[uint256(_questionID)];
+        require(arbitration.status == Status.Requested, "Invalid arbitration status");
+
+        arbitration.requester.send(arbitration.deposit);
+
+        delete arbitrations[uint256(_questionID)];
+
+        emit ArbitrationCanceled(uint256(_questionID));
+    }
+
+    /**
+     * @notice Cancels the arbitration in case the dispute could not be created.
+     * @param _arbitrationID The ID of the arbitration.
+     */
+    function handleFailedDisputeCreation(uint256 _arbitrationID) external onlyIfInitialized {
```

Comment:
Same case as above `requestArbitration` since external code depends on this method, it should be included in the interface and its signature should be:

```solidity
function handleFailedDisputeCreation(bytes32 _questionID) external
```